src/Orleans/Serialization/SerializationManager.cs

```diff
@@ -99,9 +99,11 @@ public static object GetUninitializedObjectWithFormatterServices(Type type)
         private static List<IExternalSerializer> externalSerializers;
         private static ConcurrentDictionary<Type, IExternalSerializer> typeToExternalSerializerDictionary;
         private static Dictionary<string, Type> types;
-        private static Dictionary<RuntimeTypeHandle, DeepCopier> copiers;
-        private static Dictionary<RuntimeTypeHandle, Serializer> serializers;
-        private static Dictionary<RuntimeTypeHandle, Deserializer> deserializers;
+        volatile private static Dictionary<RuntimeTypeHandle, DeepCopier> copiers;
+        volatile private static Dictionary<RuntimeTypeHandle, Serializer> serializers;
+        volatile private static Dictionary<RuntimeTypeHandle, Deserializer> deserializers;
```

Comment:
Consider using IReadonlyDictionary instead. This way since users can't mutate it by mistake unless they keep the concrete instance around (or downcast), both of which we can avoid.
Also, is the volatile keyword really necessary here? I think you should already be covered with your usage of locks for the clone and swap (lock already creates a memory barrier that forces the value to not be read from the cache).